### PR TITLE
Use ErrInvalidArgument in packages

### PR DESCRIPTION
**Pull request by KN4CK3R**
Related to https://github.com/go-gitea/gitea/pull/22262#discussion_r1059010774

### CODE REVIEW

File: routers/api/packages/rubygems/rubygems.go [hunk 1]
```diff
@@ -217,7 +218,11 @@ func UploadPackageFile(ctx *context.Context) {
 
 	rp, err := rubygems_module.ParsePackageMetaData(buf)
 	if err != nil {
-		apiError(ctx, http.StatusInternalServerError, err)
+		if errors.Is(err, util.ErrInvalidArgument) {
+			apiError(ctx, http.StatusBadRequest, err)
+		} else {
+			apiError(ctx, http.StatusInternalServerError, err)
+		}
```
Comment (delvh) [inline routers/api/packages/rubygems/rubygems.go:225]: I've seen this block now often enough.
What about an error handling function that takes `ctx` and `err` as parameters?

**Review by delvh — APPROVED** [review]

File: routers/api/packages/rubygems/rubygems.go (hunk 1 above)
Comment (KN4CK3R) [reply routers/api/packages/rubygems/rubygems.go:225]: `apiError` is always different and must then be passed as parameter too.

Comment (delvh) [reply routers/api/packages/rubygems/rubygems.go:225]: Ah, makes sense, even though I wouldn't have expected it as it always _looks_ the same.

**Review by zeripath — COMMENTED** [review]
I have a feeling we need to make a NewInvalidError function in util,

```go
// NewInvalidError returns an error that formats as the given text but unwraps as an InvalidError
func NewInvalidError(message string) error {
  return SilentWrap{Message: message, Err: util.ErrInvalidArgument}
}
```

As an aside lots of the comments on the errors have gone missing in this PR.

**Comment by zeripath** [comment]
See https://github.com/KN4CK3R/gitea/pull/4

**Comment by KN4CK3R** [comment]
> As an aside lots of the comments on the errors have gone missing in this PR.

Not missing, I intentionaly removed them on purpose. The comments were required at the time of writing and I see no benefit in comments like
```go
// ErrMissingComposerFile indicates a missing composer.json file
ErrMissingComposerFile = errors.New("composer.json file is missing")
```

**Review by lunny — APPROVED** [review]

**Review by delvh — APPROVED** [review]

**Review by lafriks — APPROVED** [review]

**Comment by KN4CK3R** [comment]
🚀

**Pull request merged**